packages/opentelemetry-sdk-trace-node/src/NodeTracerProvider.ts

```diff
@@ -68,10 +70,22 @@ export class NodeTracerProvider extends BasicTracerProvider {
     super.register(config);
   }
 
-  protected override  _getPropagator(name: string): TextMapPropagator | undefined {
+  /**
+   * TS cannot yet infer the type of this.constructor:
+   * https://github.com/Microsoft/TypeScript/issues/3841#issuecomment-337560146
+   * Do not override either of the getters in your child class unless you really need to.
+   * We have to do it here because the previous version of BasicTracerProvider's getters
+   * had BasicTracerProvider's static method hardcoded and we want do undo that behavior.
+   */
+  protected override _getPropagator(name: string): TextMapPropagator | undefined {
     return (
-      super._getPropagator(name) ||
-      NodeTracerProvider._registeredPropagators.get(name)?.()
-    );
+      (this.constructor as typeof BasicTracerProvider)._registeredPropagators
+    ).get(name)?.();
+  }
+
+  protected override _getSpanExporter(name: string): SpanExporter | undefined {
+    return (
+      (this.constructor as typeof BasicTracerProvider)._registeredExporters
+    ).get(name)?.();
```

Comment:
Actually, these two methods here can be removed as the BasicTracerProvider already finds the constructor in its implementation.

Reply:
Since it's tested and version is pinned I'm comfortable just removing it. The only way it breaks is if trace-node releases before trace-base. Since we now link in a single lerna repo, it is easier to release all together anyway.